orchagent/mplsrouteorch.cpp

```diff
@@ -421,97 +487,102 @@ bool RouteOrch::addLabelRoute(LabelRouteBulkContext& ctx, const NextHopGroupKey
 
     auto it_route = m_syncdLabelRoutes.at(vrf_id).find(label);
 
-    if (nextHops.getSize() == 0)
-    {
-        /* The route is pointing to a blackhole */
-        blackhole = true;
-    }
-    /* The route is pointing to a next hop */
-    else if (nextHops.getSize() == 1)
+    if (ctx.nhg_index.empty())
     {
-        const NextHopKey& nexthop = *nextHops.getNextHops().begin();
-        if (nexthop.isIntfNextHop())
+        if (nextHops.getSize() == 0)
         {
-            next_hop_id = m_intfsOrch->getRouterIntfsId(nexthop.alias);
-            /* rif is not created yet */
-            if (next_hop_id == SAI_NULL_OBJECT_ID)
-            {
-                SWSS_LOG_INFO("Failed to get next hop %s for %u",
-                        nextHops.to_string().c_str(), label);
-                return false;
-            }
+            /* The route is pointing to a blackhole */
+            blackhole = true;
         }
-        else
+        /* The route is pointing to a next hop */
+        else if (nextHops.getSize() == 1)
         {
-            if (m_neighOrch->hasNextHop(nexthop))
+            const NextHopKey& nexthop = *nextHops.getNextHops().begin();
+            if (nexthop.isIntfNextHop())
             {
-                next_hop_id = m_neighOrch->getNextHopId(nexthop);
-            }
-            /* For non-existent MPLS NH, check if IP neighbor NH exists */
-            else if (nexthop.isMplsNextHop() &&
-                     m_neighOrch->hasNextHop(NextHopKey(nexthop.ip_address, nexthop.alias)))
-            {
-                /* since IP neighbor NH exists, neighbor is resolved, add MPLS NH */
-                m_neighOrch->addNextHop(nexthop);
-                next_hop_id = m_neighOrch->getNextHopId(nexthop);
+                next_hop_id = m_intfsOrch->getRouterIntfsId(nexthop.alias);
+                /* rif is not created yet */
+                if (next_hop_id == SAI_NULL_OBJECT_ID)
+                {
+                    SWSS_LOG_INFO("Failed to get next hop %s for %u",
+                            nextHops.to_string().c_str(), label);
+                    return false;
+                }
             }
-            /* IP neighbor is not yet resolved */
             else
             {
-                SWSS_LOG_INFO("Failed to get next hop %s for %u",
-                        nextHops.to_string().c_str(), label);
-                return false;
+                if (m_neighOrch->hasNextHop(nexthop))
+                {
+                    next_hop_id = m_neighOrch->getNextHopId(nexthop);
+                }
+                /* See if there is an IP neighbor nexthop */
+                else if (nexthop.isMplsNextHop() &&
+                         m_neighOrch->hasNextHop(NextHopKey(nexthop.ip_address, nexthop.alias)))
```

Comment:
Change this to use nexthop.ipKey() instead?

Reply:
Fixed (but using new NeighOrch method)